enumerable_methods_ruby.rb

```diff
@@ -1 +1,132 @@
-puts "Hello world"
\ No newline at end of file
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop: disable Lint/ToEnumArguments
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    to_a.my_each do |i|
+      new_array.push(i) if yield i
+    end
+    new_array
+  end
+
+  def my_all?(param = nil)
+    if !block_given? && param.nil?
+      to_a.my_each { |i| return false if i == false || i.nil? }
+    elsif !block_given? && param.is_a?(Class)
+      to_a.my_each { |i| return false unless [i.class, i.class.superclass].include?(param) }
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return false unless i == param }
+    elsif !block_given? && (param.is_a? Regexp)
+      to_a.my_each { |i| return false unless i.match(param) }
+    elsif block_given?
+      to_a.my_each { |i| return false unless yield i }
+    end
+    true
+  end
+
+  def my_any?(param = nil)
+    if !block_given? && param.nil?
+      to_a.my_each { |i| return true if i }
+      return false
+    elsif param.is_a?(Class)
+      to_a.my_each { |i| return true if [i.class, i.class.superclass].include?(param) }
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return true if i == param }
+    elsif param.is_a?(Regexp)
+      to_a.my_each { |i| return true if i.match(param) }
+    else
+      to_a.my_each { |i| return true if yield i }
+    end
+    false
+  end
```

Comment:
- [ ] my_any? When a Regex is passed as an argument, it should return true if any of the collection matches the Regex.